diagnostic.rb

```diff
@@ -1,16 +1,37 @@
 require 'csv'
 # Diagnostic answers to:
+  rails g model License surname:string given_name:string gender:string height:integer weight:integer born_on:date eye_color:string hair_color:string
 
 # question 2:
+  caren = License.create surname: 'Bishop',
+                        given_name: 'Caren',
+                        gender: 'f',
+                        height: 63,
+                        weight: 132,
+                        born_on: 1943-09-26,
+                        eye_color: 'Brown',
+                        hair_color: 'Black'
+p caren
 
 # question 3:
-Licensee.transaction do
-  CSV.foreach 'data/licensees.csv', headers: true do |row|
+task licensees: :environment do
+  License.transaction do
+    CSV.foreach 'data/licensees.csv',
+      headers: true do |row|
+        license = row.to_hash
+        License.create! license unless License.exists?
+      end
   end
 end
 
 # question 4:
+lic = License.where 'eye_color = :eye_color AND hair_color = :h1 OR hair_color = :h2',
```

Comment:
Almost! AND has higher precedence than OR, so this works out to 
`'(eye_color = :eye_color AND hair_color = :h1) OR hair_color = :h2'`
Not quite what we wanted.